packages/angular_devkit/schematics_cli/bin/schematics.ts

```diff
@@ -38,7 +38,10 @@ function parseSchematicName(str: string | null): { collection: string, schematic
 
   let schematic = str;
   if (schematic && schematic.indexOf(':') != -1) {
-    [collection, schematic] = schematic.split(':', 2);
+    [collection, schematic] = [
+      schematic.slice(0, schematic.lastIndexOf(':')),
+      schematic.substring(schematic.lastIndexOf(':') + 1),
+    ];
```

Comment:
Thanks for this, I think this can be simplified by changing the split to use a RegExp

```suggestion
    [collection, schematic] = schematic.split(/[^A-Z]:/, 2);
```

Reply:
Pleased to help!
I've tried this before, it does split the string at the last colon but cuts the letter before it

So "C:/dir/collection.json:schematic" becomes ["C:/dir/collection.**jso**", "schematic"]

Maybe some other RegEx could fit better, but that's really not my expertise

Reply:
Actually, you could use a negative lookahead to search for the last match of ":"
Like this: **/(:)(?!.*\1)/**

But schematics.split(/(:)(?!.*\1)/), 2) will return ["C:/dir/collection.json", ":"]
Changing the limit to 3 we would have ["C:/dir/collection.json", ":", "schematics"] and then probably could use .filter or .splice to get rid of the ":" item

Don't know if it looks better with this RegEx

Reply:
Did some research, I think I've came to the right RegEx: **`/:(?=[^:]+$)/`**

```suggestion
[collection, schematic] = schematic.split(/:(?=[^:]+$)/, 2)
```

Reply:
Alright, I forgot about the `/dir/.:` case, it does not work with the regular expressions above.
Should we still try to use RegEx or settle with the `.slice()` plus `.substring()`?

Reply:
Let's go with the `slice` approach.

Reply:
Ok, all set!